### PR TITLE
Save args in config when cli_vars is saved

### PR DESCRIPTION
**Pull request by gshank**
This is for bug #4583 in dbt-core, var not injected in dbt run snapshot from dbt run invocation in 1.0.0.

### CODE REVIEW

**Review by leahwicz — APPROVED** [review]

**Comment by jtcohen6** [comment]
Looks good in local testing! I shared a Loom with my experience getting everything set up + verified.

Let me know if you need help with automated/integration testing. The RPC tests work a bit differently (and honestly better) vs. our current `dbt-core` integration testing.

After merging, let's plan to:
- Test using the `development (HEAD)` branch of dbt Cloud (includes all changes from this repo's `main` branch)
- Cut a prerelease, for use alongside `dbt-core==1.0.2rc1`
- If everything looks good, cut a final release, coordinated with `dbt-core==1.0.2` (final)

**Comment by jtcohen6** [comment]
@gshank @leahwicz Do we have a clear path forward for this one? Should we proceed with the next steps I listed above?

**Review by ChenyuLInx — CHANGES_REQUESTED** [review]
We probably should have a test to test this behavior

File: tests/test_management.py [hunk 1]
```diff
@@ -361,3 +361,53 @@ def test_get_manifest(
...
+    with querier_ctx as querier:
+      results = querier.async_wait_for_result(querier.cli_args('run --vars \'test_variable: "1234"\''))
+      assert len(results['results']) == 1
+      assert results['results'][0]['node']['compiled_sql'] == 'select 1234 as id'
+
+
+config_var_sql = '''
+{{ config(unique_key=var('test_variable')) }}
+select {{ config.get('unique_key') }} as id
+'''
```
Comment (stu-k) [inline tests/test_management.py:391]: I'm not sure if this method of testing that the config is as expected is too roundabout, I can modify if it is.

File: tests/test_management.py [hunk 2]
```diff
@@ -361,3 +361,53 @@ def test_get_manifest(
...
+        project_def=project,
+        project_dir=project_root,
+        profiles_dir=profiles_root,
+        schema=unique_schema,
+    )
+
+    with querier_ctx as querier:
+      results = querier.async_wait_for_result(querier.cli_args('run --vars \'test_variable: "9876"\''))
+      assert len(results['results']) == 1
+      assert results['results'][0]['node']['compiled_sql'].strip() == 'select 9876 as id'
```
Comment (stu-k) [inline tests/test_management.py:413]: The compiled sql produces a newline at the beginning of the string, `strip`ping the string removes this.

**Review by ChenyuLInx — APPROVED** [review]
LGTM!

**Pull request merged**